be/src/exec/vectorized/sorting/sorting.h

```diff
@@ -50,5 +52,92 @@ void build_tie_for_column(const ColumnPtr column, Tie* tie);
 // Append rows from permutation
 void append_by_permutation(Column* dst, const Columns& columns, const Permutation& perm);
 void append_by_permutation(Chunk* dst, const std::vector<ChunkPtr>& chunks, const Permutation& perm);
+void append_by_permutation(Chunk* dst, const std::vector<const Chunk*>& chunks, const Permutation& perm);
+
+struct SortDesc {
+    int sort_order;
+    int null_first;
+};
+
+struct SortDescs {
+    std::vector<int> sort_orders;
+    std::vector<int> null_firsts;
+
+    SortDescs() = default;
+    ~SortDescs() = default;
+    SortDescs(const std::vector<int>& orders, const std::vector<int>& nulls)
+            : sort_orders(orders), null_firsts(nulls) {}
+
+    size_t num_columns() const { return sort_orders.size(); }
+
+    SortDesc get_column_desc(int col) const {
+        SortDesc desc;
+        DCHECK_LT(col, sort_orders.size());
+        desc.sort_order = sort_orders[col];
+        desc.null_first = null_firsts[col];
+        return desc;
+    }
+};
+
+struct SortedRun {
```

Comment:
would better add some comments for this struct.

Reply:
done